File: translated/f-to-c/DRB080-func-arg-orig-yes.c
/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Copyright (c) 2017-20, Lawrence Livermore National Security, LLC
and DataRaceBench project contributors. See the DataRaceBench/COPYRIGHT file for details.

SPDX-License-Identifier: (BSD-3-Clause)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/

// argument pass-by-reference
// its data-sharing attribute is the same as its actual argument's. i and q are shared.
// Data race pair: q@15:5:W vs. q@15:5:W

#include <omp.h>
#include <stdio.h>

void f1(int* q)
{
    *q = *q + 1;
}

int main()
{
    int i;

    i = 0;

    #pragma omp parallel
    f1(&i);

    printf("i = %d\n", i);

    return 0;
} 

File: translated/f-to-go-automodel-bad/DRB080-func-arg-orig-yes.go
//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
// Copyright (c) 2017-20, Lawrence Livermore National Security, LLC
// and DataRaceBench project contributors. See the DataRaceBench/COPYRIGHT file for details.
//
// SPDX-License-Identifier: (BSD-3-Clause)
//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

// argument pass-by-reference
// its data-sharing attribute is the same as its actual argument's. i and q are shared.
// Data race pair: q@15:5:W vs. q@15:5:W

package main

import (
	"fmt"
	"sync"
)

func f1(q *int) {
	*q = *q + 1
}

func main() {
	var i int

	i = 0

	//$omp parallel
	var wg sync.WaitGroup
	wg.Add(1)
	go func() {
		defer wg.Done()
		f1(&i)
	}()
	wg.Wait()
	//$omp end parallel

	fmt.Printf("i = %d\n", i)
}
